OMEdit/OMEditLIB/OMS/ElementPropertiesDialog.cpp

```diff
@@ -439,57 +439,47 @@ bool ElementPropertiesDialog::eventFilter(QObject *pObject, QEvent *pEvent)
  * helper function to restore default start values read from modeldescription.xml for fmus
  * and 0 for other systems
  */
-void ElementPropertiesDialog::restoreDefaultStartValue(QString pName, QLineEdit * pLineEdit)
+void ElementPropertiesDialog::deleteStartValueAndRestoreDefault(QString pName, QLineEdit * pLineEdit)
 {
   if (mpComponent->getLibraryTreeItem()->getOMSElement() && mpComponent->getLibraryTreeItem()->getOMSElement()->connectors) {
     oms_connector_t** pInterfaces = mpComponent->getLibraryTreeItem()->getOMSElement()->connectors;
     for (int i = 0 ; pInterfaces[i] ; i++) {
       QString nameStructure = QString("%1.%2").arg(mpComponent->getLibraryTreeItem()->getNameStructure(), QString(pInterfaces[i]->name));
-      if (pInterfaces[i]->causality == oms_causality_parameter && QString(pInterfaces[i]->name)== pName) {
-        OMSProxy::instance()->omsDelete(nameStructure + ":start");
-        restoreDefaultStartValueHelper(pInterfaces[i], nameStructure, pLineEdit);
-        return;
-      } else if (pInterfaces[i]->causality == oms_causality_input && QString(pInterfaces[i]->name)== pName) {
-        OMSProxy::instance()->omsDelete(nameStructure + ":start");
-        restoreDefaultStartValueHelper(pInterfaces[i], nameStructure, pLineEdit);
-        return;
+      if (QString(pInterfaces[i]->name).compare(pName) == 0) {
+        if (oms_causality_parameter == pInterfaces[i]->causality || oms_causality_input == pInterfaces[i]->causality) {
```

Comment:
Make it a one if condition,
`if ((QString(pInterfaces[i]->name).compare(pName) == 0) && (oms_causality_parameter == pInterfaces[i]->causality || oms_causality_input == pInterfaces[i]->causality)) {`

Reply:
i think the condition can be at two levels, otherwise the line becomes too lengthy to read

Reply:
I agree with @arun3688. It becomes very cumbersome to read if it is in one line.

Reply:
Can there be multiple pInterfaces[i]s with the same name? or is it unique names?

If they are unique names then this should probably return after matching a name but not matching causality.